### ISSUE TITLE
Feature request:

### ISSUE THREAD

**Issue by Barben360** [benjaminadk/emojigit#5]
Hello it's me again and again!
Would it be possible to add the possibility to add comments to the commit?

From git commit documentation:

> -m <msg>, --message=<msg>
           Use the given <msg> as the commit message. If multiple -m options
           are given, their values are concatenated as separate paragraphs.

This means that comments could be appended in the command line reusing -m "comment line 1" -m "comment line 2", etc

**Comment by Barben360** [comment]
However this inserts a double-backline between each paragraph, which is not exactly a multiline comment.

**Comment by Barben360** [comment]
Yes I'm thinking about this. However I've just tested, using \n character doesn't go back to line, it just inserts characters \ and n in the commit title.

**Comment by Barben360** [comment]
Some hint here: `git commit -F <filename>` creates the commit message from a file. Therefore if you generate a temp file with the commit title and message (including multiline), you can just generate the temp file and generate the line:
`git commit -F <filename> && rm <filename>`. However the rm will only work on unix-based distributions. I think you should consider applying the commit and removing the file directly in your extension and not just generate the command line.

**Comment by benjaminadk** [comment]
So does the user create the temp file in vscode? Or do I parse the user input from the vscode UI and generate a file from that? I have to look into vscode api and see if it offers a multiline input. Right now all I can think of is parsing the single line text for the `\n` character and formatting a template string and creating the temp file with the extension.

**Comment by Barben360** [comment]
As a user I think it is a bad idea to make the user create the temp file. This is quite a lot of work for just a commit. Generating it would be the best. If multiline input is possible in quick actions, then this would be the best solution. I have no idea if this is possible, but I think not. Your solution parsing \n would be quite a good compromise I think.
Another solution would be to have a section in "Source control" panel on the left to manage gitmoji commits, multiline in forms is possible here (at least it is for the built-in git message editor). But that would imply a full redesign of your extension.

**Comment by akaleeroy** [comment]
First of all nice work on the extension! :clap: 

I'd like multi-line commits too. I'm coming from using a [custom git commit template](https://git-scm.com/book/en/v2/Customizing-Git-Git-Configuration#_code_commit_template_code):

```
git config --global commit.template ~/.gitmessage.txt
```

`.gitmessage.txt` file ↓

```
# https://gitmoji.carloscuesta.me

# :art: Improving structure / format of the code
# :zap: Improving performance
# :fire: Removing code or files
# ...
```

Committing looks like:
`git commit -a` 
↳ `COMMIT_EDITMSG` file opens in default editor
↳ track down what kind of commit I want, un-comment it, write my commit message, hit Return twice, write the details, save and close

It's a great experience because you have the full power of a code editor, you get multi-line commits, spell checking, you can set rulers to 50 characters, you get another look at the committed file names and you can change your mind: _Hang on this isn't a :pencil2: **Fixing typos** it's a :speech_balloon: **Changing text and literals...**_

The upsides with your extension are triggering with a command (incl. keybinding) and that sweet sweet fuzzy search command palette menu, instead of doing <kbd>Ctrl</kbd> <kbd>F</kbd>. I can't figure out how to integrate the commit template workflow directly into VSCode, save for opening a terminal and typing out `git commit`. If you try to use the Source control panel it forces you into a little textbox.

**Comment by benjaminadk** [comment]
I am brainstorming. I just found [this article](https://coderwall.com/p/-rlneg/multi-line-git-commit-message-from-cli) and the last option - adding `$` between `-m` and the quoted message. The article says that `$` makes the shell honor `\n` on `*nix` systems. This works on my Windows 10 machine. This could be an option is vscode settings. Lile `allowNewline` or something. 

---

OR - maybe try to integrate with git's flow somehow. If extensions could write to user's machine i guess it could overwrite the template and open the editor with the selected emoji and text on the subject line on each commit. not sure if this is possible or the best approach though.

**Comment by Barben360** [comment]
Nice discovery! I also found out that you can commit directly a text file as a message:

```
git commit -F message.txt
```

In the documentation of VSCode extensions, I can see several interesting things:

- [Virtual documents](https://code.visualstudio.com/api/extension-guides/virtual-documents) to generate read-only files (maybe the commit message can be one of them?)
- [Source control API](https://code.visualstudio.com/api/extension-guides/scm-provider) which allows you to integrate with source control (and then probably git flow).

**Comment by benjaminadk** [comment]
I added basic multiline support. 
Having a problem with my azure devops account. I didnt renew an email address i used for the microsoft account i created the account for this extension with. 
Bottom line is that i cant publish an update at this point. I am looking into it.

**Comment by pke** [comment]
@benjaminadk can you submit updates by now again?
